crates/ty_server/src/session.rs

```diff
@@ -266,21 +270,30 @@ impl Session {
                         metadata.apply_overrides(overrides);
                     }
 
-                    ProjectDatabase::new(metadata, system)
-                        .context("Failed to create project database")
+                    ProjectDatabase::new(metadata, system.clone())
                 });
 
-            // TODO(micha): Handle the case where the program settings are incorrect more gracefully.
-            // The easiest is to ignore those projects but to show a message to the user that we do so.
-            // Ignoring the projects has the effect that we'll use the default project for those files.
-            // The only challenge with this is that we need to register the project when the configuration
-            // becomes valid again. But that's a case we need to handle anyway for good mono repository support.
             match project {
                 Ok(project) => {
                     self.projects.insert(root, project);
                 }
                 Err(err) => {
-                    tracing::warn!("Failed to create project database for `{root}`: {err}",);
+                    tracing::error!("Failed to create project for `{root}`: {err:#}");
+                    tracing::info!("Falling back to default settings");
```

Comment:
I think these should be in the same message. It might be a bit confusing to see the "Falling back ..." message without understanding that it's tied to the previous message.